src/controllers/movies-controller.ts

```diff
@@ -0,0 +1,96 @@
+import { NextFunction, Request, Response } from 'express';
+import { IError } from '../error-handlers/general-error-handler.js';
+
+import { IMovie, Movie } from '../models/Movie.js';
+
+const getAllMovies = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const movies = await Movie.find();
+    res.status(200).json(movies);
+  } catch (error) {
+    next(error);
+  }
+};
+
+const addMovie = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    if (!req.body.title) {
+      const error: IError = new Error('Movie title is required.');
+      error.statusCode = 400;
+      error.status = 'failed';
+      throw error;
+    }
+    if (!req.body.description) {
+      const error: IError = new Error('Movie description is required.');
+      error.statusCode = 400;
+      error.status = 'failed';
+      throw error;
+    }
+    if (!req.body.releaseDate) {
+      const error: IError = new Error('Please, enter the release date');
+      error.statusCode = 400;
+      error.status = 'failed';
+      throw error;
+    }
+    if (!req.body.genre) {
+      const error: IError = new Error('Please, enter movie genre');
+      error.statusCode = 400;
+      error.status = 'failed';
+      throw error;
+    }
+    const movie: IMovie = await Movie.create({
+      title: req.body.title,
+      description: req.body.description,
+      releaseDate: req.body.releaseDate,
+      genre: req.body.genre,
+    });
```

Comment:
Consider moving such checks to validation middleware.

Reply:
Should I use mongoose validation if I use validation middleware? Can I leave it or it will be too much?

Reply:
You can leave both because these validation have different purposes. Keep in mind that if you validate same field in two places with same criteria, better to use some shared files with constant values, so you not apply different criteria to same data.

Reply:
Thanks!